src/moin/items/__init__.py

```diff
@@ -674,7 +674,8 @@ def _save(self, meta, data=None, name=None, action=ACTION_SAVE, contenttype_gues
             meta[REV_NUMBER] = 1
 
         if not overwrite and REVID in meta:
-            # we usually want to create a new revision, thus we must remove the existing REVID
+            # we usually want to create a new revision, thus we update parentid and remove the existing REVID
+            meta[PARENTID] = currentrev.meta[REVID] if currentrev else meta[REVID]
             del meta[REVID]
```

Comment:
looks good for the case of an existing ("on disk") currentrev.

but what happens if currentrev is falsy - then we keep its revid as parentid, kill the revid from meta (which causes the storage to generate a new revid, right?). what is the meaning of the parentid then?

Reply:
The only case I could find where currentrev is false is for a new item and then "REVID in meta" is also false so meta[PARENTID] is not set because the if fails.

I think currentrev will always be true when the if succeeds and I could not find a case where it was not true. The fallback to meta[REVID] was to avoid a traceback just in case.

Reply:
we can keep that as is for now.